addons/hr_work_entry_contract/models/hr_contract.py

```diff
@@ -100,6 +101,35 @@ def _get_attendance_intervals(self, start_dt, end_dt):
             ))
         return result
 
+    def _postprocess_attendance_intervals(self, intervals):
+        # _attendance_intervals_batch combines the attendances regardless of work entry type if their date and time overlap.
+        # This makes a single attendance block with the work entry type of the first attendance only. This function undoes those
+        # undesirable merges
+        # intervals: [(start_date, end_date, attendances)]
+        try:
+            multi_type_intervals = [interval for interval in intervals._items if len(interval[2].mapped('work_entry_type_id')) > 1]
+        except KeyError:
```

Comment:
I've had cases where work_entry_type_id is not found. Not entirely sure where they came from but it seems safer to leave that in to me

Reply:
If the contract's work entry source is 'Planning' this happens. We have planning.slot instead of attendances which do not have work_entry_type_id.

![image](https://github.com/user-attachments/assets/67a2553c-e334-4b75-abd3-fbb03df6e395)
![image](https://github.com/user-attachments/assets/244ceb02-e8e5-4b4d-a8d7-44d338588df4)

Reply:
I guess this might be the only case, I could check for it instead of having a try-except.